generator/generator.go

```diff
@@ -20,13 +20,38 @@ type featureHelpers struct {
 	feature int
 }
 
-type ObjectSet map[protogen.GoIdent]bool
+type ObjectSet struct {
+	wildcards map[string]struct{}
+	mp        map[protogen.GoIdent]bool
+}
+
+func NewObjectSet() ObjectSet {
+	return ObjectSet{
+		mp:        map[protogen.GoIdent]bool{},
+		wildcards: map[string]struct{}{},
+	}
+}
 
 func (o ObjectSet) String() string {
 	return fmt.Sprintf("%#v", o)
 }
 
+func (o ObjectSet) Contains(g protogen.GoIdent) bool {
+	for wildcard := range o.wildcards {
+		if Match(wildcard, fmt.Sprintf("%s.%s", string(g.GoImportPath), g.GoName)) {
```

Comment:
I think we'd be better set by using the standard library's `path.Match`, since the `GoImportPath` here contains slashes, it'd be good if our wildcard function was slash-aware.

Also, why the composition with `Sprintf` here? Doesn't `protogen.GoIdent` have a `String` method? Pretty sure it does.

Reply:
Agreed, should use path.Match instead.

`protogen.GoIdent` have `String` method. But uses quoted `GoImportPath` part.
(https://github.com/protocolbuffers/protobuf-go/blob/master/compiler/protogen/protogen.go#L1226)